File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/atoms/placeholder/roles/ShapePlaceholderRole.kt
package com.perrystreet.woof.designsystem.atomic.atoms.placeholder.roles

import androidx.compose.runtime.Composable
import androidx.compose.ui.unit.Dp
import com.perrystreet.woof.designsystem.theme.Theme

enum class ShapePlaceholderRole {
    Image {
        @Composable
        override fun radius(): Dp = Theme.radius.s

        @Composable
        override fun aspectRatio(): Float? = null
    },

    Card {
        @Composable
        override fun radius(): Dp = Theme.radius.s

        @Composable
        override fun aspectRatio(): Float = Theme.aspectRatios.gridCell
    },
    ;

    @Composable
    abstract fun radius(): Dp

    @Composable
    abstract fun aspectRatio(): Float?
}

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Atoms/Placeholder/Roles/ShapePlaceholderRole.swift
import SwiftUI

public enum ShapePlaceholderRole {
    case image
    case card

    public func radius(from theme: ThemeImplementing) -> CGFloat {
        switch self {
        case .image, .card: theme.radius.s
        }
    }

    public func aspectRatio(from theme: ThemeImplementing) -> CGFloat? {
        switch self {
        case .image: nil
        case .card: theme.aspectRatios.gridCell
        }
    }
}
